### PR TITLE
Update kotlin to 2.0.20; Add client request retry plugin

### PR DESCRIPTION
**Pull request by e5l**
Update kotlin to 2.0.20 and add client retry plugin to fix some flaky tests

### CODE REVIEW

**Review by bjhham — APPROVED** [review]
🙏

**Comment by e5l** [comment]
I will wait for the CI build to check it's not worse 😁

**Comment by bjhham** [comment]
The anticipation is killing me 😅

File: buildSrc/src/main/kotlin/Compilations.kt [hunk 1]
```diff
@@ -2,16 +2,11 @@
...
-//        if (platformType == KotlinPlatformType.jvm && !IDEA_ACTIVE) {
-//            allWarningsAsErrors = true
-//        }
-
-        freeCompilerArgs += "-opt-in=kotlin.RequiresOptIn"
-        freeCompilerArgs += "-Xexpect-actual-classes"
+fun KotlinCompilationTask<*>.configureCompilerOptions() {
+    compilerOptions {
+        progressiveMode.set(true)
+        freeCompilerArgs.addAll("-opt-in=kotlin.RequiresOptIn", "-Xexpect-actual-classes")
```
Comment (Goooler) [inline buildSrc/src/main/kotlin/Compilations.kt:10]:
```suggestion
        freeCompilerArgs.addAll("-Xexpect-actual-classes")
```

https://kotlinlang.org/docs/whatsnew17.html#stable-opt-in-requirements

File: build.gradle.kts [hunk 2]
```diff
@@ -192,12 +189,11 @@ fun Project.setupJvmToolchain() {
...
-    targets.all {
-        compilations.all {
-            configureCompilation()
-        }
-    }
+subprojects {
+    val compileKotlin = tasks.findByName("compileKotlin") as? KotlinCompilationTask<*>
+        ?: return@subprojects
+
+    compileKotlin.configureCompilerOptions()
```
Comment (Goooler) [inline build.gradle.kts:196]:
```suggestion
    tasks.withType<KotlinCompilationTask<*>>().configureEach {
        configureCompilerOptions()
    }
```

Comment (Goooler) [reply build.gradle.kts:196]: And there is already `subprojects` below, we can reuse it.

Comment (e5l) [reply build.gradle.kts:196]: thanks!

File: build.gradle.kts [hunk 3]
```diff
@@ -2,11 +2,9 @@
  * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
  */
 
+import org.gradle.api.Project
```
Comment (Goooler) [inline build.gradle.kts:5]:
```suggestion
```

File: build.gradle.kts [hunk 4]
```diff
@@ -192,11 +189,9 @@ fun Project.setupJvmToolchain() {
     }
 }
 
-fun KotlinMultiplatformExtension.setCompilationOptions() {
-    targets.all {
-        compilations.all {
-            configureCompilation()
-        }
+subprojects {
```
Comment (Goooler) [inline build.gradle.kts:192]: Please reuse the `subprojects` below.

**Pull request merged**